### PR TITLE
12 add capability information automatically on build from the standard app api

### LINKED ISSUE #12: Add capability information automatically on build from the standard app API

### PR DESCRIPTION
**Pull request by pablo-de-andres**
This PR closes #12.

The capability table is now automatically created by processing the openAPI.json from the [standard app api repo](https://github.com/materials-marketplace/standard-app-api) with the version defined in `utils.py`.
Redoc also renders said file now.

The sample files have removed, since they would require manual maintenance.
Instead, users should be pointed to the get-started app once it is updated

### CODE REVIEW

**Comment by pablo-de-andres** [comment]
Some notes:
- I would not merge this until the next release of the standard app api, since the release currently used (0.1.0) is quite limited.
- @csadorf in the capability table I used the "description" field of the openAPI.json (seemed more meaningful than the "summary". However, long descriptions (in `newTransformation` and `updateTransformation` capabilities) do not look too nice (I had to convert them from multi-line to single line so the rendering was not broken). Should I use another field? Assume the first line is the relevant one and ignore the rest?
image

**Comment by csadorf** [comment]
> * @csadorf in the capability table I used the "description" field of the openAPI.json (seemed more meaningful than the "summary". However, long descriptions (in `newTransformation` and `updateTransformation` capabilities) do not look too nice (I had to convert them from multi-line to single line so the rendering was not broken). Should I use another field? Assume the first line is the relevant one and ignore the rest?

@pablo-de-andres I think we should use the ["summary" field](https://spec.openapis.org/oas/v3.1.0#fixed-fields-6) which is currently auto-generated as far as I can tell, but can be defined [like this](https://fastapi.tiangolo.com/tutorial/path-operation-configuration/?h=summary#summary-and-description).

**Comment by pablo-de-andres** [comment]
Ok. I will update the code here and make a PR in the standard app API repo to start adding them.

File: docs/source/index.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # MarketPlace Docs
 
-Welcome to the [MarketPlace](https://www.materials-marketplace.eu/) documentation!
+Welcome to the [MarketPlace](https://www.materials-marketplace.eu/) documentation (version {platform_version})!
```
Comment (yoavnash) [inline docs/source/index.md]: I think it doesn't look so good. Maybe we mention the version in  a separate line or even in the footer?

File: docs/source/apps/mp-api.md [hunk 2]
```diff
@@ -3,9 +3,12 @@
 To declare which capabilities are supported by your application, a REST-API that complies with the MarketPlace specification must be set for your application.
 The API should be expressed using the [OpenAPI specification](https://www.openapis.org/) standard and be sent to the MarketPlace platform upon [registration](registration.md).
 
+You can browse the official specification with redoc <a href="api.html" target="_blank">here</a>.
+The complete specification is hosted publicly on [GitHub](https://github.com/materials-marketplace/standard-app-api).
+
 ## OpenAPI specification
 
-In this section we will explain some of the most relevant parts of the specification and provide examples for easier development.
+In this section we will explain some of the most relevant parts of the specification.
```
Comment (yoavnash) [inline docs/source/apps/mp-api.md]: I think it's more standard to use simple present tense: (I added another change on the way)
```suggestion
In this section we explain some of the non-trivial parts of the specification.
```

File: docs/source/index.md (hunk 1 above)
Comment (pablo-de-andres) [reply docs/source/index.md]: Do you think an admonition at the top (like info or so) would be better?
It seems there might be [some issues](https://github.com/executablebooks/jupyter-book/issues/1660) with templates, and we cannot customise the footer.

Comment (yoavnash) [reply docs/source/index.md]: Yeah, maybe that could work better.

Comment (pablo-de-andres) [reply docs/source/index.md]: Ok, I applied the changes. Feel free to re-review.

**Review by yoavnash — APPROVED** [review]

**Pull request merged**